src/main/c/functions.c

```diff
@@ -1,44 +1,159 @@
 #include <math.h>
 #include <stdlib.h>
+#include <string.h> // For memset.
 #include "data.h"
 #include "functions.h"
 
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
 /**
  * Returns the size of a triangular matrix, without the diagonal.
  */
-size_t size_triangular_matrix(const size_t n) {
+size_t
+size_triangular_matrix(const size_t n) {
   return ((n - 1) * n) / 2;
 }
 
+/**
+ * Computes the distance between two particles.
+ */
+double
```

Comment:
The function definition is recommended to be split in this way.
`void
func_name()
{}`
So it is easier to search for function declarations, as stated in https://www.gnu.org/prep/standards/standards.html#index-line-length.

Reply:
I already give my two cents about this on the previous comment but anyways.

IMHO, it looks horrible and it is not easier to read. But, that is just an opinion, as already said, it should be chosen by all the team.

Second, the argument about searching does not hold any more.
Nowadays we have IDEs that have _go-to_ definition.
And, our code base is too short and well organized to make it difficult to search.